$NetBSD: patch-aa,v 1.31 2013/01/12 11:55:35 jmcneill Exp $

o Handle properly sh3 and evbarm ports on NetBSD.
o Fix CPU family detection on NetBSD with /proc mounted with "-o linux"
  (please see also kern/25076).
o Fix a bug which broke the detection of a lot of X11 libraries.
o Add necessary libs for aalib support even if the aa option is turned on with
  --enable-aa.
o Add support for dvb capture on NetBSD.

--- configure.orig	2012-05-22 18:01:51.058982400 +0000
+++ configure
@@ -1505,6 +1505,9 @@ if test -z "$_target" ; then
   OS/2*)
     system_name=OS/2
     ;;
+  Haiku*)
+    system_name=Haiku
+    ;;
   *)
     system_name="$system_name-UNKNOWN"
     ;;
@@ -1517,10 +1520,12 @@ if test -z "$_target" ; then
       ia64) host_arch=ia64 ;;
       macppc|ppc*|Power*) host_arch=ppc ;;
       alpha) host_arch=alpha ;;
+      BePC) host_arch=x86 ;;
       sun4*|sparc*) host_arch=sparc ;;
       parisc*|hppa*|9000*) host_arch=hppa ;;
-      arm*|zaurus|cats) host_arch=arm ;;
-      sh3|sh4|sh4a) host_arch=sh ;;
+      arm*|zaurus|cats|evbarm) host_arch=arm ;;
+      sh3*) host_arch=sh ;;
+	  sh4|sh4a) host_arch=sh4 ;;
       s390) host_arch=s390 ;;
       s390x) host_arch=s390x ;;
       *mips*) host_arch=mips ;;
@@ -1674,7 +1679,7 @@ else
       cc_name=$cc_name_tmp
       echocheck "$_cc version"
       cc_vendor=gnu
-      cc_version=$($_cc -dumpversion 2>&1)
+      cc_version=$( ( $_cc -dumpversion | sed -e 's/-nb[0-9]//' ) 2>&1 )
       case $cc_version in
         2.96*)
           cc_fail=yes
@@ -1798,7 +1803,7 @@ if x86 ; then
   # gather more CPU information
   pname=$($_cpuinfo | grep 'model name' | cut -d ':' -f 2 | head -n 1)
   pvendor=$($_cpuinfo | grep 'vendor_id' | cut -d ':' -f 2  | cut -d ' ' -f 2 | head -n 1)
-  pfamily=$($_cpuinfo | grep 'cpu family' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
+  pfamily=$($_cpuinfo | grep -i 'cpu family' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
   pmodel=$($_cpuinfo | grep -v 'model name' | grep 'model' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
   pstepping=$($_cpuinfo | grep 'stepping' | cut -d ':' -f 2 | cut -d ' ' -f 2 | head -n 1)
 
@@ -2274,7 +2279,12 @@ case "$host_arch" in
     test $_fast_clz = "auto" && _fast_clz=yes
     ;;
 
-  sh|sh4)
+  sh)
+    arch='sh'
+    iproc='sh'
+    ;;
+
+  sh4)
     arch='sh4'
     iproc='sh4'
     ;;
@@ -2591,7 +2601,7 @@ fi # if darwin && test "$cc_vendor" = "g
 # Checking for CFLAGS
 _install_strip="-s"
 if test "$_profile" != "" || test "$_debug" != "" ; then
-  CFLAGS="-O2 $_march $_mcpu $_pipe $_debug $_profile"
+  CFLAGS="$CFLAGS $_debug $_profile"
   WARNFLAGS="-W -Wall"
   _install_strip=
 elif test -z "$CFLAGS" ; then
@@ -4288,7 +4298,11 @@ if test "$_x11" = auto && test "$_x11_he
            -L/usr/X11/lib -L/usr/lib32 -L/usr/openwin/lib -L/usr/local/lib64 -L/usr/X11R6/lib64 \
            -L/usr/lib ; do
     if netbsd; then
-      ld_tmp="$I -lXext -lX11 $ld_pthread -Wl,-R$(echo $I | sed s/^-L//)"
+	  if test -n "$I" ; then
+        ld_tmp="$I -lXext -lX11 $ld_pthread -Wl,-R$(echo $I | sed s/^-L//)"
+	  else
+        ld_tmp="-lXext -lX11 $ld_pthread"
+	  fi
     else
       ld_tmp="$I -lXext -lX11 $ld_pthread"
     fi
@@ -4669,6 +4683,8 @@ if test "$_aa" = yes ; then
   def_aa='#define CONFIG_AA 1'
   if cygwin ; then
     libs_mplayer="$libs_mplayer $(aalib-config --libs | cut -d " " -f 2,5,6)"
+  else
+    libs_mplayer="$libs_mplayer $(aalib-config --libs)"
   fi
   vomodules="aa $vomodules"
 else
@@ -4756,7 +4772,6 @@ EOF
       extra_cflags="$extra_cflags $inc_tmp" && break
   done
 fi
-echores "$_dvb"
 if test "$_dvb" = yes ; then
   _dvbin=yes
   inputmodules="dvb $inputmodules"
@@ -4765,14 +4780,29 @@ if test "$_dvb" = yes ; then
   aomodules="mpegpes(dvb) $aomodules"
   vomodules="mpegpes(dvb) $vomodules"
 else
-  _dvbin=no
-  noinputmodules="dvb $noinputmodules"
-  def_dvb='#undef CONFIG_DVB'
-  def_dvbin='#undef CONFIG_DVBIN '
-  aomodules="mpegpes(file) $aomodules"
-  vomodules="mpegpes(file) $vomodules"
+  cat >$TMPC <<EOF
+#include <dev/dtv/dtvio.h>
+int main(void) {return 0;}
+EOF
+  cc_check && _dvb=yes
+  if test "$_dvb" = yes ; then
+    _dvbin=yes
+    inputmodules="dvb $inputmodules"
+    def_dvb='#undef CONFIG_DVB'
+    def_dvbin='#define CONFIG_DVBIN 1'
+    aomodules="mpegpes(file) $aomodules"
+    vomodules="mpegpes(file) $vomodules"
+  else
+    _dvbin=no
+    noinputmodules="dvb $noinputmodules"
+    def_dvb='#undef CONFIG_DVB'
+    def_dvbin='#undef CONFIG_DVBIN '
+    aomodules="mpegpes(file) $aomodules"
+    vomodules="mpegpes(file) $vomodules"
+  fi
 fi
 
+echores "$_dvb"
 
 if darwin; then
 
